### PR TITLE
[image_picker] Do not resize GIF images

### PR DESCRIPTION
**Pull request by ened**
## Description

Reenables image picking for .GIF images on Android instead of converting anything to JPG straight away.

## Related Issues

Closes https://github.com/flutter/flutter/issues/34134.
Closes https://github.com/flutter/flutter/issues/52419.
Closes https://github.com/flutter/flutter/issues/88461.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by ened** [comment]
This works a little different on Android than on iOS. There seems to be a gif scaler on iOS, whereas I'm not sure it exists anywhere in the Android SDK.

@cyanglaz do you have a suggestion how to test this PR? There are different paths for GIF and other image formats. I was thinking to just extend the FileUtilTest and perhaps check if the response from one of the methods matches the original path, if that is somehow possible to determine.

**Comment by ditman** [comment]
Is this related to https://github.com/flutter/plugins/pull/2982? They seem similar?

**Comment by ened** [comment]
@ditman yes - 2982 solves one problem (the returned file extension being jpg for any file) while this PR solves two:
 - file extension always jpg
 - GIF images are resized which stops the animation on Android

We can either merge 2982 first or confirm this PR. @ponnamkarthik do you think this PR contains the same fix as yours from a technical point of view?

**Comment by ponnamkarthik** [comment]
@ditman you can merge @ened PR my PR is similar, I have runned ened code and it fix the problem with that fixes my PR

**Comment by ditman** [comment]
I found another implementation of this fix in https://github.com/flutter/plugins/pull/2860 (only the JPG extension part), can we merge these two/grab ideas from both? I've noticed the extension detection seems a little bit more thorough in 2860 than here.

File: packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java [hunk 1]
```diff
@@ -557,7 +567,7 @@ private void handleImageResult(String path, boolean shouldDeleteOriginalIfScaled
 
       finishWithSuccess(finalImagePath);
 
-      //delete original file if scaled
+      // delete original file if scaled
```
Comment (pedromassango) [inline packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java:570]: I think this sentence should start with upper case letter!

**Comment by TahaTesser** [comment]
@ened 
Can you please separate the extension from resizing GIF fix and update with rebase, new null-safety version
Thank you

**Comment by ened** [comment]
@TahaTesser could do - but what are the odds of this getting merged soon?

**Comment by TahaTesser** [comment]
@ened 
Tag me, I will follow up with you and try to get it to merge if the PR is approved, thank you

**Comment by ened** [comment]
@TahaTesser this should build now.

**Comment by TahaTesser** [comment]
@ened 
Thanks, I will check it soon

**Comment by ditman** [comment]
The broken `build-apks+java-test+firebase-test-lab CHANNEL:stable` check seems to be a timeout downloading stuff. I've restarted it.

**Comment by stuartmorgan** [comment]
Removing `last mile` since this isn't approved yet. @ditman were you planning to review this?

**Comment by ditman** [comment]
Ah shoot, no @stuartmorgan, I was using 'last mile' as a flag for "this is probably ready to merge, somebody take a look". My bad.

**Comment by stuartmorgan** [comment]
Thanks for the contribution!

> * [x]  My PR includes unit or integration tests for _all_ changed/updated/fixed behaviors (See [Contributor Guide](https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md)).

I don't see any tests in this PR. Before we would love forward with reviewing, it would need tests, per the linked policy.

After that we can try to take a look relatively soon since this is a fairly small PR, but we can't promise a specific timeline; we're working through a large backlog, and issue votes are the primary prioritization method for PRs, as described in the contributing guide.

**Comment by ditman** [comment]
@googlebot I consent.

**Comment by ditman** [comment]
I've tweaked this a little bit, and added tests. @ened can you confirm this still works for you with my small change?

**Comment by ditman** [comment]
And broken the unit tests... hmmm....

**Comment by ened** [comment]
@ditman lgtm

**Comment by ditman** [comment]
Hey all, I tried to detect if an image is animated or not, but I broke one of my tests (which should pass after my change) and couldn't setup a decent Android environment to debug the test. Here's my latest change in case somebody wants to keep at it:

https://github.com/ditman/plugins/commit/e0bddc50aaa5e26fd397be4940c0e48091f22ea5

**Comment by stuartmorgan** [comment]
Are you still planning on updating this PR based on the last round of discussion?

**Comment by ditman** [comment]
@stuartmorgan I haven't had time to look at this in a few months, somebody with more android experience should tackle it as suggested by @cyanglaz, probably!

**Comment by stuartmorgan** [comment]
@ened I see you have updated the branch, but it doesn't look like you've addressed the comments. Is this still something you're planning on finishing?

**Comment by stuartmorgan** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**